runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,420 @@
+		{
+			log.debug("Not connected to a chat yet.", e);
+		}
+	}
+
+	//Triggers when a message is sent to twitch chat
+	@Handler
+	public void onMessage(ChannelMessageEvent a)
+	{
+		String username = a.getActor().getUserString();
+
+		//Sub length badges
+		//0 months
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/0"))
+		{
+			username = "<img=9>" + username;
+		}
+		//3 months
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/3"))
+		{
+			username = "<img=8>" + username;
+		}
+		//6 months
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/6"))
+		{
+			username = "<img=7>" + username;
+		}
+		//12 months
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/12"))
+		{
+			username = "<img=6>" + username;
+		}
+		//24 months
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/24"))
+		{
+			username = "<img=5>" + username;
+		}
+		//48 months
+		else if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("subscriber/48"))
+		{
+			username = "<img=4>" + username;
+		}
+
+		//Moderator badge
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("moderator/1"))
+		{
+			username = "<img=26>" + username;
+		}
+
+		//Streamer badge
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("broadcaster/1"))
+		{
+			username = "<img=27>" + username;
+		}
+
+		//Show the message in game with color based on if the chat is transparent or not
+		String message = a.getMessage();
+		if (isChatboxTransparent() && client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+		else if (client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=000000>[</col><col=0000FF>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+	}
+
+	//User and room information update events
+	//Triggers whenever the user's state is updated or they join a room
+	@Handler
+	public void onUserStateEvent(UserStateEvent u)
+	{
+		String originalMessage = u.getOriginalMessage().toString();
+		badges = "";
+
+		//Finding if the user is a subscriber
+		if ((originalMessage.contains("subscriber=1") || originalMessage.contains("mod=1") || originalMessage.contains("value=broadcaster/1")) && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			subscribed = true;
+			if (originalMessage.contains("subscriber=1"))
+			{
+				if (originalMessage.contains("subscriber/0"))
+				{
+					badges = "<img=9>" + badges;
```

Comment:
Why the `badges = ... + badges` in all the if statements in "subscriber=1"? `badges` is an empty string at this point. `badges = "<img=...>"` should be enough.